tests/run_tests/driver_mag_test.py

```diff
@@ -0,0 +1,246 @@
+    octree.add_data({"model": {"values": model[mesh._ubc_order]}})
+    octree.copy()  # Keep a copy around for ref
+
+    return workspace
+
+
+def test_susceptibility_run(tmp_path):
+    from geoapps.drivers.magnetic_scalar_inversion import MagneticScalarDriver
+    from geoapps.io.MagneticScalar.params import MagneticScalarParams
+
+    # Inducing field
+    H0 = (50000.0, 90.0, 0.0)
+    # Values stored from pre-runs
+    target = {
+        "data_norm": 716.999,
+        "phi_m": 0.006225,
+        "phi_d": 92500.0,
+    }
+    # Run the forward
+    workspace = setup_workspace(tmp_path, 0.05)
+    model = workspace.get_entity("model")[0]
+    params = MagneticScalarParams(
+        forward=True,
+        workspace=workspace,
+        mesh=model.parent,
+        topography_object=workspace.get_entity("topography")[0],
+        inducing_field_strength=H0[0],
+        inducing_field_inclination=H0[1],
+        inducing_field_declination=H0[2],
+        resolution=0.0,
+        data_object=workspace.get_entity("survey")[0],
+        starting_model_object=model.parent,
+        starting_model=model,
+    )
+    driver = MagneticScalarDriver(params)
+    driver.run()
+
+    workspace = Workspace(workspace.h5file)
+    tmi = workspace.get_entity("tmi")[0]
+    np.testing.assert_almost_equal(
+        np.linalg.norm(tmi.values), target["data_norm"], decimal=3
+    )
+
+    # Run the inverse
+    np.random.seed(0)
+    params = MagneticScalarParams(
+        workspace=workspace,
+        mesh=workspace.get_entity("mesh")[0],
+        topography_object=workspace.get_entity("topography")[0],
+        inducing_field_strength=H0[0],
+        inducing_field_inclination=H0[1],
+        inducing_field_declination=H0[2],
+        resolution=0.0,
+        data_object=tmi.parent,
+        starting_model=1e-4,
+        s_norm=0.0,
+        x_norm=0.0,
+        y_norm=0.0,
+        z_norm=0.0,
+        gradient_type="components",
+        lower_bound=0.0,
+        tmi_channel_bool=True,
+        z_from_topo=False,
+        tmi_channel=tmi,
+        tmi_uncertainty=1.0,
+        max_iterations=1,
+    )
+    driver = MagneticScalarDriver(params)
+    driver.run()
+
+    # Re-open the workspace and get iterations
+    output = get_inversion_output(
+        driver.params.workspace.h5file, driver.params.out_group.uid
+    )
+    np.testing.assert_almost_equal(output["phi_m"][1], target["phi_m"])
+    np.testing.assert_almost_equal(output["phi_d"][1], target["phi_d"])
+
+    ##########################################################################
+    # Solution should satisfy this condition if run to completion.
+    # To get validated with full run if values in 'target' need to be updated.
+    #
+    # residual = (
+    #         np.linalg.norm(driver.inverse_problem.model - model) /
+    #         np.linalg.norm(model) * 100.
+    # )
+    # assert residual < 0.1, (
+    #     f"Deviation from the true solution is {residual}%. Please revise."
```

Comment:
Nice touch!  I guess you can just run this a bunch of times in debug mode??  This doesn't look like it will run as is though, right?  driver doesn't contain a .inverse_problem attribute - is there a model in the output dictionary we can use in place of this?

Reply:
Actually it doesn't work for another reason - model here is a FloatData object and needs to be .value'ed, active_cell mapped, nan removed, and possibly reordered.